common/persistence/client/factory.go

```diff
@@ -261,7 +261,7 @@ func (f *factoryImpl) Close() {
 
 func IsPersistenceTransientError(err error) bool {
 	switch err.(type) {
-	case *serviceerror.Unavailable:
+	case *serviceerror.Unavailable, *serviceerror.DataLoss:
```

Comment:
can you add some comments here re. why we are doing this?

Reply:
Done.